rfcs/proposed/zip_view/README.md

```diff
@@ -0,0 +1,40 @@
+# zip_view Support for the oneDPL Range APIs with C++20
+
+## Introduction
+`std::ranges::zip_view` is a powerful utility that enables developers to combine two or more ranges into a single view,
+where each element is represented as a tuple containing corresponding elements from each input range.
+
+## Motivations
+`std::ranges::zip_view` is a convenient way to combine multiple ranges into a single view, where each element of
+the resulting range is a tuple containing one element from each of the input ranges. This can be particularly
+useful for iterating over multiple collections in parallel. `std::ranges::zip_view` was introduced in C++23,
+but many developers are still using C++20 standard. So, oneDPL introduces `oneapi::dpl::ranges::zip_view`,
+with the same API and functionality as `std::ranges::zip_view`.
+
+### Key Requirements
+`oneapi::dpl::ranges::zip_view` should be:
+- compilable with C++20 version (minimum)
+- API-compliant with `std::ranges::zip_view`
+- random accessible view; the "underlying" views also should be random accessible
+- in case of a device usage: a device copyable view if the all "underlying" views are device copyable views
+  
+`oneapi::dpl::ranges::zip_view::iterator` should be:
+- value-swappable (https://en.cppreference.com/w/cpp/named_req/ValueSwappable)
+- implicitly convertible to `oneapi::dpl::zip_iterator`
+- able to be used with the non-range algorithms
+
+### Implementation proposal
+- `oneapi::dpl::ranges::zip_view` is designed as a C++ class which represents a range adaptor (see C++ Range Library).
+This class encapsulates a tuple-like type to keep a combination of two or more ranges.
+- The implementation provides all necessary operators to satisfy 'random accessible view' requirement.
+- To ensure device copyability, `oneapi::dpl::__internal::tuple` is proposed as a tuple-like type for underlying elements.
+- To provide a value-swappable requirement `oneapi::dpl::__internal::tuple` is proposed as a dereferenced value for
```

Comment:
When it comes to the specification, this will need to be described as an unspecified tuple which satisfies some set of requirements, unless we want to make the move to specify our internal tuple and make it public (which I doubt we want to do).

Reply:
In the context of this, I was looking at the specification for our current `zip_iterator`, and noticed a couple of discrepancies as compared to our implementation with regard to std::tuple vs our unspecified internal tuple type.  I believe that our implementation is intentional and the "correct" thing here and the spec should be changed, because of issues with device copyability when composing types from `zip_iterators`.  `std::tuple` is not trivially copyable but our internal tuple is.  

https://github.com/uxlfoundation/oneAPI-spec/pull/605

Reply:
We are moving towards specifying the tuple-like type better, most likely as a type that satisfies certain concepts / requirements. That should be done in an new proposal though.

Reply:
Yes.. I've mentioned it in the RFC document update.